R/redist_ms_tidy.R

```diff
@@ -0,0 +1,173 @@
+#####################################################
+# Author: Cory McCartan
+# Institution: Harvard University
+# Date Created: 2021/01/31
+# Purpose: tidy R wrapper to run Merge-Split/Recom redistricting code
+####################################################
+
+#' Merge-Split/Recombination MCMC Redistricting Sampler
+#'
+#' \code{redist_mergesplit} uses a Markov Chain Monte Carlo algorithm to
+#' generate congressional or legislative redistricting plans according to
+#' contiguity, population, compactness, and administrative boundary constraints.
+#' The MCMC proposal is the same as is used in the SMC sampler; it is similar
+#' but not identical to those used in the references.
+#'
+#' This function draws samples from a specific target measure, controlled by the
+#' \code{popcons}, \code{compactness}, \code{constraints}, and
+#' \code{constraint_fn} parameters.
+#'
+#' Higher values of \code{compactness} sample more compact districts;
+#' setting this parameter to 1 is computationally efficient and generates nicely
+#' compact districts.
+#'
+#' The \code{constraints} parameter allows the user to apply several common
+#' redistricting constraints without implementing them by hand. This parameter
+#' is a list, which may contain any of the following named entries:
+#' * \code{status_quo}: a list with two entries:
+#'   * \code{strength}, a number controlling the tendency of the generated districts
+#'   to respect the status quo, with higher values preferring more similar
+#'   districts.
+#'   * \code{current}, a vector containing district assignments for
+#'   the current map.
+#' * \code{vra}: a list with five entries:
+#'   * \code{strength}, a number controlling the strength of the Voting Rights Act
+#'   (VRA) constraint, with higher values prioritizing majority-minority districts
+#'   over other considerations.
+#'   * \code{tgt_vra_min}, the target percentage of minority voters in minority
+#'   opportunity districts. Defaults to 0.55.
+#'   * \code{tgt_vra_other} The target percentage of minority voters in other
+#'   districts. Defaults to 0.25, but should be set to reflect the total minority
+#'   population in the state.
+#'   * \code{pow_vra}, which controls the allowed deviation from the target
+#'   minority percentage; higher values are more tolerant. Defaults to 1.5
+#'   * \code{min_pop}, A vector containing the minority population of each
+#'   geographic unit.
+#' * \code{incumbency}: a list with two entries:
+#'   * \code{strength}, a number controlling the tendency of the generated districts
+#'   to avoid pairing up incumbents.
+#'   * \code{incumbents}, a vector of precinct indices, one for each incumbent's
+#'   home address.
+#'
+#'
+#' @param map A \code{\link{redist_map}} object.
```

Comment:
I might be blind, but I can't find a pop_tol constraint here.

Reply:
Right, it's embedded inside the `redist_map` object